codalab/apps/web/static/js/worksheet/worksheet_interface.js

```diff
@@ -148,28 +152,26 @@ var Worksheet = React.createClass({
           $("#raw-textarea").focus();
         }
     },
-    toggleActionBar: function(){
-        this.setState({showActionBar:!this.state.showActionBar});
+    toggleActionBar: function() {
+        this.setState({showActionBar: !this.state.showActionBar});
     },
-    showActionBar: function(){
-        this.setState({showActionBar:true});
+    showActionBar: function() {
+        this.setState({showActionBar: true});
     },
-    hideActionBar: function(){
-        this.setState({showActionBar:false});
+    hideActionBar: function() {
+        this.setState({showActionBar: false});
     },
-    refreshWorksheet: function(){
+    refreshWorksheet: function() {
         $('#update_progress').show();
-        this.setState({updating:true});
+        this.setState({updating: true});
         ws_obj.fetch({
-            success: function(data){
-                // console.log("fetch_and_update success");
-                // console.log("%c--------------------------------------------------------------", "color: Green; font-size:15px;");
-                if(this.isMounted()){
+            success: function(data) {
+                if (this.isMounted()) {
                     this.refs.list.setState({worksheet: ws_obj.getState()});
```

Comment:
Yes, this is definitely not good, and I've been trying to fix this so that
they are more React-like.  Updated the code with a TODO.  Let's get this PR
checked in soon (it's already grown a bit out of control), so we can build
on top of it.

On Sun, Oct 18, 2015 at 10:53 PM, Pujun Bhatnagar <notifications@github.com>
wrote:

> In codalab/apps/web/static/js/worksheet/worksheet_interface.js
> <https://github.com/codalab/codalab/pull/1342#discussion_r42337967>:
>
> >          ws_obj.fetch({
> > -            success: function(data){
> > -                // console.log("fetch_and_update success");
> > -                // console.log("%c--------------------------------------------------------------", "color: Green; font-size:15px;");
> > -                if(this.isMounted()){
> > +            success: function(data) {
> > +                if (this.isMounted()) {
> >                      this.refs.list.setState({worksheet: ws_obj.getState()});
>
> This may be an anti pattern for React. If we are setting the state of
> anything other than self in a component's class, that indicates that the
> source data is not stored at one place and we have multiple copies which is
> not ideal when the virtual DOM is created and the diffing happens.
>
> However, to fix this, we would have to do some major re-factoring of the
> code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codalab/codalab/pull/1342/files#r42337967>.
>